rdt/hyper_transformer.py

```diff
@@ -691,8 +691,19 @@ def fit(self, data):
         self._validate_detect_config_called(data)
         self._unfit()
         self._input_columns = list(data.columns)
-        for field_column, field_transformer in self.field_transformers.items():
-            data = self._fit_field_transformer(data, field_column, field_transformer)
+        skipped_columns = []  # skip columns in multi column transformer already fitted
+        for column in self._input_columns:
```

Comment:
any reason to loop over this instead of the `field_transformers`?

Reply:
Before defining the `multi-column transformers` we were iterating over `self._input_columns`. I found that this order can be important sometimes for the constraint, and that's why some tests were failing. By iterating over field_transformers, some randomness in the order makes some tests fail sometimes.